src/baldr/graphtile.cc

```diff
@@ -148,19 +148,18 @@ GraphTile GraphTile::CacheTileURL(const std::string& tile_url,
 
   // If its good try to use it
   if (http_code == 200) {
-    // try to cache it on disk so we dont have to keep doing this
+    // try to cache it on disk so we dont have to keep fetching it from url
     if (!cache_location.empty()) {
       auto disk_location = cache_location + filesystem::path::preferred_separator + suffix;
       auto dir = filesystem::path(cache_location + filesystem::path::preferred_separator + suffix);
       dir.replace_filename("");
       if (filesystem::create_directories(dir)) {
         std::ofstream file(disk_location, std::ios::out | std::ios::binary | std::ios::ate);
         file.write(&tile_data[0], tile_data.size());
-        return GraphTile(cache_location, graphid);
```

Comment:
this was a bug and a slower way of doing it. it was a bug because we didnt call `file.close()` before teling the constructor to go read it off the disk. it was slow because we already have the bytes in memory from curl and so we just load them directly below